frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/ol-map-interactions-manager.ts

```diff
@@ -0,0 +1,173 @@
+        private readonly store: Store<AppState>,
+        private readonly popupManager: PopupManager,
+        private readonly olMap: OlMap,
+        private readonly layerFeatureManagerDictionary: Map<
+            VectorLayer<VectorSource>,
+            FeatureManager<any>
+        >,
+        private readonly destroy$: Subject<void>
+    ) {
+        this.featureLayers = [];
+        this.trainerInteractions = [];
+        this.interactions = new Collection<Interaction>();
+        this.translateInteraction = new TranslateInteraction();
+        this.participantInteractions = [this.translateInteraction];
+        this.updateInteractions();
+        this.registerInteractionEnablementHandler();
+    }
+
+    public addFeatureLayer(layer: VectorLayer<VectorSource>) {
+        this.featureLayers.push(layer);
+        this.updateRegisterAndApplyAll();
+    }
+
+    public addTrainerInteraction(interaction: Interaction) {
+        this.trainerInteractions.push(interaction);
+        this.updateRegisterAndApplyAll();
+    }
+
+    private updateRegisterAndApplyAll() {
+        this.updateInteractions();
+        this.registerDropHandler();
+        this.applyInteractions();
+        this.updateInteractionEnablement(this.lastStatus, this.lastRole);
+    }
+
+    private updateTranslateInteraction() {
+        this.translateInteraction = new TranslateInteraction({
+            layers: this.featureLayers,
+            hitTolerance: 10,
+            filter: (feature, layer) => {
+                const featureManager = this.layerFeatureManagerDictionary.get(
+                    layer as VectorLayer<VectorSource>
+                );
+                return featureManager === undefined
+                    ? false
+                    : featureManager.isFeatureTranslatable(feature);
+            },
+        });
+    }
+
+    private updateParticipantInteractions() {
+        this.participantInteractions = [this.translateInteraction];
+    }
+
+    private updateInteractions() {
+        this.updateTranslateInteraction();
+        this.updateParticipantInteractions();
+        this.interactions = defaultInteractions({
+            pinchRotate: false,
+            altShiftDragRotate: false,
+            keyboard: true,
+        }).extend(
+            selectStateSnapshot(selectCurrentRole, this.store) === 'trainer'
+                ? [...this.participantInteractions, ...this.trainerInteractions]
+                : [...this.participantInteractions]
+        );
+    }
+
+    public applyInteractions() {
+        this.mapInteractions.clear();
+        // We just want to modify this for the Map not do anything with it after so we ignore the returned value
+        // eslint-disable-next-line rxjs/no-ignored-observable
+        this.mapInteractions.extend(this.interactions.getArray());
+    }
+
+    // Register handlers that disable or enable certain interactions
+    registerInteractionEnablementHandler() {
```

Comment:
Is this method supposed to be private or public?

Reply:
Private + Done